gridcapa-swe-runner-app/src/main/java/com/farao_community/farao/swe/runner/app/services/CneFileExportService.java

```diff
@@ -0,0 +1,129 @@
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+package com.farao_community.farao.swe.runner.app.services;
+
+import com.farao_community.farao.commons.FaraoException;
+import com.farao_community.farao.data.cne_exporter_commons.CneExporterParameters;
+import com.farao_community.farao.data.crac_creation.creator.cim.crac_creator.CimCracCreationContext;
+import com.farao_community.farao.data.rao_result_api.RaoResult;
+import com.farao_community.farao.data.swe_cne_exporter.SweCneExporter;
+import com.farao_community.farao.minio_adapter.starter.MinioAdapter;
+import com.farao_community.farao.rao_api.parameters.RaoParameters;
+import com.farao_community.farao.swe.runner.api.exception.SweInvalidDataException;
+import com.farao_community.farao.swe.runner.api.resource.ProcessType;
+import com.farao_community.farao.swe.runner.app.dichotomy.DichotomyDirection;
+import com.farao_community.farao.swe.runner.app.domain.SweData;
+import com.powsybl.commons.datasource.MemDataSource;
+import org.springframework.stereotype.Service;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.time.OffsetDateTime;
+import java.time.ZoneOffset;
+import java.time.format.DateTimeFormatter;
+import java.util.UUID;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipOutputStream;
+
+/**
+ * @author Marc Schwitzguébel {@literal <marc.schwitzguebel at rte-france.com>}
+ */
+
+@Service
+public class CneFileExportService {
+
+    public static final String FILENAME_TIMESTAMP_REGEX = "yyyyMMdd'_'HHmm'_'";
+    public static final String TIME_INTERVAL_REGEX = "yyyy'-'MM'-'dd'T'HH':00:00Z'";
+
+    private final FileExporter fileExporter;
+    private final MinioAdapter minioAdapter;
+
+    public CneFileExportService(FileExporter fileExporter, MinioAdapter minioAdapter) {
+        this.fileExporter = fileExporter;
+        this.minioAdapter = minioAdapter;
+    }
+
+    public String exportCneUrl(SweData sweData, RaoResult raoResult, boolean isHighestValid, ProcessType processType, DichotomyDirection direction) {
+
+        OffsetDateTime timestamp = sweData.getTimestamp();
+        // limit size to 35 characters, a UUID is 36 characters long
+        String mRid =  UUID.randomUUID().toString().substring(1);
+        CneExporterParameters cneExporterParameters = new CneExporterParameters(
+                mRid, 1, "", CneExporterParameters.ProcessType.Z01,
+                "10XFR-RTE------Q", CneExporterParameters.RoleType.SYSTEM_OPERATOR,
+                "22XCORESO------S", CneExporterParameters.RoleType.CAPACITY_COORDINATOR,
+                extractTimeIntervalFileHeader(timestamp));
+
+        CimCracCreationContext cracCreationContext = sweData.getCracFrEs();
+        if (direction == DichotomyDirection.ES_PT || direction == DichotomyDirection.PT_ES) {
```

Comment:
those 4 lines looks a bit silly. You can exctract it in a method et testing the 4 directions and return  what is needed